src/components/input/input.js

```diff
@@ -322,7 +322,9 @@ function inputTextareaDirective($mdUtil, $window, $mdAria) {
             node.style.minHeight = null;
           }
 
-          var rows = Math.max(min_rows, Math.round(node.scrollHeight / lineHeight));
+          var rows = Math.min(min_rows, Math.round(node.scrollHeight / lineHeight));
+          node.style.height = lineHeight * rows + "px";
+
           node.setAttribute("rows", rows);
```

Comment:
Why did you change Math.max to Math.min? I have my textarea set to rows="3" and want it to stay at a minimum of 3 rows, growing if necessary. With your Math.min, the "min_rows" variable (and hence the "rows" attribute) basically becomes "max_rows", which seems unintended.